src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,200 @@
+                    break;
+                } catch (MissingFormatArgumentException ex) {
+                    ui.printMissingArgumentError(keyword);
+                    break;
+                } catch (DukeException ex) {
+                    ui.printMissingDateTimeArgumentError(keyword);
+                    break;
+                } catch (IOException e) {
+                    ui.printWriteError();
+                    break;
+                }
+            }
+            case "event": {
+                try {
+                    if (tokens.length <= 1) {
+                        throw new MissingFormatArgumentException("no argument detected");
+                    }
+                    cmd.event(tokens, history);
+                    storage.update(history);
+                    break;
+                } catch (MissingFormatArgumentException ex) {
+                    ui.printMissingArgumentError(keyword);
+                    break;
+                } catch (DukeException ex) {
+                    ui.printMissingDateTimeArgumentError(keyword);
+                    break;
+                } catch (IOException e) {
+                    ui.printWriteError();
+                    break;
+                }
+            }
+            case "delete": {
+                try {
+                    int index = Integer.parseInt(tokens[1]);
+                    cmd.delete(index - 1, history);
+                    storage.update(history);
+                    break;
+                } catch (NumberFormatException | NullPointerException | IndexOutOfBoundsException ex) {
+                    ui.printInvalidArgumentError();
+                    break;
+                } catch (IOException e) {
+                    ui.printWriteError();
+                    break;
+                }
+            }
+            case "update": {
+                try {
+                    storage.update(history);
+                    break;
+                } catch (IOException e) {
+                    ui.printWriteError();
+                    break;
+                }
+            }
+            case "find": {
+                try {
+                    if (tokens.length <= 1) {
+                        throw new MissingFormatArgumentException("no argument detected");
+                    }
+                    cmd.find(tokens, history);
+                    break;
+                } catch (MissingFormatArgumentException ex) {
+                    ui.printMissingArgumentError(keyword);
+                    break;
+                } catch (DukeException e) {
+                    ui.printFoundNothing();
+                    break;
+                }
+            }
+            default:
+                try {
+                    throw new MissingFormatArgumentException("invalid keywords");
+                } catch (MissingFormatArgumentException ex) {
+                    ui.printListOfCommands();
+                    break;
+                }
+            }
```

Comment:
I like how you managed to account for the different commands in `run()` using a `switch` statement, would it be better for `run()` to have a layer of abstraction such as a `Parser` to handle the different cases in the `switch` statement?